### PR TITLE
Add memcache

### PR DESCRIPTION
**Pull request by PaulKovalov**
Add class which handles cache operations

### CODE REVIEW

File: sticknotesbackend/src/main/java/com/google/sticknotesbackend/FastStorage.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    Cache cache = getCacheInstance();
+    String cacheKey = Long.toString(boardId);
+    Whiteboard board = (Whiteboard)cache.get(cacheKey);
+    if (board == null) {
+      // if cache is empty, load from datastore
+      Key<Whiteboard> boardKey = Key.create(Whiteboard.class, boardId);
+      board = ofy().load().group(Whiteboard.WithoutNotesAndCreator.class).key(boardKey).now();
+      // save board in cache for future use
+      cache.put(cacheKey, board);
+      return board;
```
Comment (AndrzejPodo) [inline sticknotesbackend/src/main/java/com/google/sticknotesbackend/FastStorage.java:69]: this line is not needed, right?

File: sticknotesbackend/src/main/java/com/google/sticknotesbackend/FastStorage.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  /**
+   * Returns a UserBoardRole object for given board and user with given google acc id.
+   * Tries to get value from cache, if it is empty gets value from datastore and updates cache
+   */
+  public static UserBoardRole getUserBoardRole(Long boardId, String googleAccId) {
+    Cache cache = getCacheInstance();
+    String cacheKey = Long.toString(boardId) + "-" + googleAccId;
+    UserBoardRole role = (UserBoardRole)cache.get(cacheKey);
+    if (role == null) {
+      // permission is not in cache yet, so load it and put into cache
```
Comment (AndrzejPodo) [inline sticknotesbackend/src/main/java/com/google/sticknotesbackend/FastStorage.java:83]: "permission"? Haven't you meant "role"?

**Review by AndrzejPodo — APPROVED** [review]

File: sticknotesbackend/src/main/java/com/google/sticknotesbackend/FastStorage.java (hunk 1 above)
Comment (PaulKovalov) [reply sticknotesbackend/src/main/java/com/google/sticknotesbackend/FastStorage.java:69]: right

File: sticknotesbackend/src/main/java/com/google/sticknotesbackend/FastStorage.java (hunk 2 above)
Comment (PaulKovalov) [reply sticknotesbackend/src/main/java/com/google/sticknotesbackend/FastStorage.java:83]: yes, it was role firstly and I forgot to update comment

**Pull request merged**